src/darwin/Framework/CHIP/MTROTAProviderDelegateBridge.mm

```diff
@@ -608,81 +608,82 @@ bool GetPeerNodeInfo(CommandHandler * commandHandler, const ConcreteCommandPath
+                      if (![commandParams.protocolsSupported
+                              containsObject:@(MTROtaSoftwareUpdateProviderOTADownloadProtocolBDXSynchronous)]) {
+                          response.status
+                              = static_cast<OTAQueryStatus>(MTROTASoftwareUpdateProviderOTAQueryStatusDownloadProtocolNotSupported);
+                      } else if ([data.status isEqual:@(MTROtaSoftwareUpdateProviderOTAQueryStatusUpdateAvailable)]) {
+                          ConvertFromQueryImageResponseParms(data, response);
+
+                          // If there is an update available, return a response with status - Available and image uri set in the
+                          // response
+                          auto fabricIndex = handler->GetSubjectDescriptor().fabricIndex;
+                          auto nodeId = handler->GetSubjectDescriptor().subject;
+                          CHIP_ERROR err = gOtaSender.PrepareForTransfer(fabricIndex, nodeId);
+                          if (CHIP_NO_ERROR != err) {
+                              LogErrorOnFailure(err);
+
+                              // Handle busy error separately as we have a query image response status that maps to busy
+                              if (err == CHIP_ERROR_BUSY) {
+                                  response.status = static_cast<OTAQueryStatus>(MTROTASoftwareUpdateProviderOTAQueryStatusBusy);
+                                  if (!response.delayedActionTime.HasValue()) {
+                                      response.delayedActionTime.SetValue(kDelayedActionTimeSeconds);
+                                  }
+                              } else {
+                                  handler->AddStatus(cachedCommandPath, StatusIB(err).mStatus);
+                                  handle.Release();
+                                  gOtaSender.ResetState();
+                                  return;
+                              }
+                          }
+                          auto targetNodeId
+                              = handler->GetExchangeContext()->GetSessionHandle()->AsSecureSession()->GetLocalScopedNodeId();
+
+                          char uriBuffer[kMaxBDXURILen];
+                          MutableCharSpan uri(uriBuffer);
+                          err = bdx::MakeURI(targetNodeId.GetNodeId(), AsCharSpan(data.imageURI), uri);
+                          if (CHIP_NO_ERROR != err) {
+                              LogErrorOnFailure(err);
+                              handler->AddStatus(cachedCommandPath, StatusIB(err).mStatus);
+                              handle.Release();
+                              gOtaSender.ResetState();
+                              return;
+                          }
+
+                          response.imageURI.SetValue(uri);
+                          handler->AddResponse(cachedCommandPath, response);
+                          handle.Release();
```

Comment:
So if we get here...  Won't we crash on the `handler->AddResponse` bits below, since the handler is gone now?  Again, we really need tests.

Reply:
need to remove this. my bad

Reply:
re-architected this bit as requested